### DISCUSSION TITLE
Provide a brew tap to install gdg

### DISCUSSION THREAD

**Discussion by iptizer** [esnet/gdg#90] [Q&A (Help Section)] [answered]
**Is your feature request related to a problem? Please describe.**
Gdg must be compiled and cannot be installed via brew.

**Describe the solution you'd like**
Provide a tap to install gdg directly via brew on mac. See [brew docs](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap).

**Describe alternatives you've considered**
-

**Additional context**
Maybe I find time to create a PR, but would be interesting whether this was already thought of or why this does not yet exist.

**Comment by safaci2000** [comment]
Mainly I'm not familiar with taps outside of using them as a user.  I'm not opposed to having a tap but I'm not sure I'd like to maintain it.  

Currently releases are fairly trivial with just pushing a tag causes the CI/CD process to create artifacts which you can download with a releases for virtually every architecture/OS you can think of.  (https://github.com/esnet/gdg/releases)

My issue with adding brew is that it's a slippery slope to start supporting other packages like (.deb, .rpm, dmg, flakpack etc).  I'd much rather not have to manage all these package managers unless there's a strong advantage you can see that isn't solved by simply either running:

`go install https://github.com/esnet/gdg@latest` or the release pages gives precompiled binaries with sample configs.

My biggest argument against it is more of a maintenance overhead rather than the initial work to figure it out.

**Comment by safaci2000** [comment, marked as answer]
I looked into this a bit more, it seems like there is a way of doing this with goreleaser which removes the maintenance aspect.  I'll need to spend some time on this but: https://github.com/esnet/gdg/issues/89 captures that.  If you want to try to play around with goreleaser in a fork, I'll happily create a tap repo

**Reply by safaci2000** [reply]
This is now done.  Feel free to install using a homebrew tap.  Instructions are available here:  https://software.es.net/gdg/docs/gdg/installation/#homebrew-installation
